src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -29,27 +29,41 @@ import {
   supportsPolygon,
   supportsThorchain,
 } from '@shapeshiftoss/hdwallet-core'
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
 import { MetaMaskShapeShiftMultiChainHDWallet } from '@shapeshiftoss/hdwallet-shapeshift-multichain'
 import { useMemo } from 'react'
 import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { selectAccountIdsByChainId } from 'state/slices/portfolioSlice/selectors'
+import { useAppSelector } from 'state/store'
 
 type WalletSupportsChainArgs = {
   isSnapInstalled: boolean | null
+  chainAccountIds: AccountId[] // allows dynamic chain-support detection for Ledger
```

Comment:
The guts of this PR

Reply:
MetaMask, too?

```suggestion
  chainAccountIds: AccountId[] // allows dynamic chain-support detection for Ledger and MetaMask
```

Reply:
Not quite, for MM we're still relying on the `isSnapInstalled` property!